viewer/source/assets/shaders/outlines.vert.glsl

```diff
@@ -0,0 +1,51 @@
+#version 450
+
+layout(location = 0) in vec4 inPosition;
+layout(location = 1) in vec2 inOffset;
+layout(location = 2) in vec2 inSize;
+layout(location = 3) in vec3 inInfo;
+
+layout(set = 1, binding = 0) uniform Params {
+    float resolution;
+    float maxDepth;
+} params;
+
+layout(location = 0) out vec3 outInfo;
+
+void main() {
+    float lineWidth = 1.0 / params.resolution;
+
+    if (inInfo.y < 3) {
+        lineWidth *= 8.0 - inInfo.y;
+    } else if (inInfo.y < 5) {
+        lineWidth *= 5.0 - inInfo.y;
+    }{
+        lineWidth *= 1;
+    }
+
+    vec2 nodePosition = inPosition.xy;
+
+    if (inPosition.w == 0.0) {
```

Comment:
I'm passing four quads and scale & move them to be the top, right, bottom, left border based on the w-component rather than passing an actual line geometry.
Seems not clean to me?